test.spec.js

```diff
@@ -29,6 +32,27 @@ t.test('basic async hash', t => {
   });
 }).catch(t.threw);
 
+t.test('async hash with null in password', t => {
+  'use strict';
+
+  t.plan(1);
+
+  return argon2.hash('pass\0word', salt).then(hash => {
+    t.equal(hash, '$argon2i$m=4096,t=3,p=1$c29tZXNhbHQ$tcauj48oAe6NE/VLzTawLTQtmX848wkNs1d7z53ahNE');
+  });
+}).catch(t.threw);
+
+t.test('async hash with null in salt', t => {
+  'use strict';
+
+  t.plan(1);
+
+  return argon2.hash(password, saltWithNull).then(hash => {
+    const saltBase64 = hash.substring(24, 46);
+    t.equal(saltBase64, saltWithNullBase64);
+  });
```

Comment:
I prefer if you use `saltBase64.toString('base64')` instead since it is semantically better.